web/client/src/library/components/sourceList/SourceList.tsx

```diff
@@ -36,13 +39,16 @@ export default function SourceList<
   to: string
   items?: TItem[]
   types?: TType
+  activeItemIndex?: number
   byName?: string
   disabled?: boolean
   byDescription?: string
   className?: string
 }): JSX.Element {
   const [filter, setFilter] = useState('')
 
+  const scrollableAreaRef = useRef<HTMLDivElement>(null)
+
   const filtered =
```

Comment:
i think we can have this return tuple: `active index` and `filtered list` (just list if no `filter` text provided)